bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,410 @@
+                            valueConfigurationToTransformed);
+                    break;
+                case SecondGenerationBindingConstants.CHANNEL_MAXDEPTHOFDISCHARGESET:
+                    valueConfiguration = command.toString();
+                    dxsEntriesConf = "33556247";
+                    preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                            valueConfiguration);
+                    break;
+                case SecondGenerationBindingConstants.CHANNEL_SHADOWMANAGEMENTSET:
+                    valueConfiguration = command.toString();
+                    dxsEntriesConf = "33556483";
+                    preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                            valueConfiguration);
+                    break;
+                case SecondGenerationBindingConstants.CHANNEL_EXTERNALMODULECONTROLSET:
+                    valueConfiguration = command.toString();
+                    dxsEntriesConf = "33556482";
+                    preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                            valueConfiguration);
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set channel configuration parameters
+        channelConfigs = SecondGenerationChannelConfiguration.getChannelConfiguration();
+        channelConfigsExt = SecondGenerationChannelConfiguration.getChannelConfigurationExt();
+        channelConfigsExtExt = SecondGenerationChannelConfiguration.getChannelConfigurationExtExt();
+        channelConfigsConfigurable = SecondGenerationChannelConfiguration.getChannelConfigurationConfigurable();
+
+        // Set inverter configuration parameters
+        final SecondGenerationInverterConfig inverterConfig = getConfigAs(SecondGenerationInverterConfig.class);
+        this.inverterConfig = inverterConfig;
+
+        // Initialize Gson object gson
+        final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        this.gson = gson;
+
+        // Temporary value during initializing
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Start update as configured
+        secondGenerationPoller = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                refresh();
+                updateStatus(ThingStatus.ONLINE);
+            } catch (RuntimeException scheduleWithFixedDelayException) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        scheduleWithFixedDelayException.getClass().getName() + ":"
+                                + scheduleWithFixedDelayException.getMessage());
+                logger.debug("Error when connecting to inverter, {}: {}", getThing().getUID(),
+                        scheduleWithFixedDelayException.getMessage());
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.